### PR TITLE
Version upgrades and cleanups

### PR DESCRIPTION
**Pull request by briancappello**
This PR aims to bring the BlogBundle more in line with the [CMF Bundle Standards](http://symfony.com/doc/current/cmf/contributing/bundles.html).

##### General Changes
* Updated dependencies to ``CMF 1.2.*``
* Updated routing configuration to work with ``RoutingAutoBundle 1.0.*``
* Parameter and service ID names have been changed to match the conventions laid out in the standards
* DI in general got cleaned up with the intent of making it easier for 3rd party bundles to extend the BlogBundle

##### Persistence
* ``Blog`` and ``Post`` have been refactored into the Model and Doctrine/Phpcr namespaces.
* Their PHPCR mapping files were likewise relocated into Resources/config/doctrine-phpcr
* The mapping files are now auto-loaded by a ``DoctrinePhpcrMappingsPass``, if the class is found
* Added a ``GenericInitializer`` service to create the ``cmf_blog.blog_basepath`` if it's not found
* Rewrote the ``PostRepository`` to support searching by an array input-verified with the OptionsResolver

##### Front End
* Separated ``BlogController`` into ``BlogController`` and ``PostController``
* Templates were moved, renamed and refactored accordingly. Added markup for easier styling (still on BS2)

##### New Features
* Added (optional) route to list blogs
* Added (optional) pagination support, using the ``KnpPaginatorBundle``
* Added (optional) PHPCR data fixture to ease development

##### Known Issues
* Make sure configuration is loadable when defined as XML
* The XML schema file for the config is out-of-date with Configuration
* Translatable content. It might work; untested.

Addresses issues #53 and #57

### CODE REVIEW

File: Repository/PostRepository.php [hunk 1]
```diff
@@ -11,29 +11,141 @@
...
 
 namespace Symfony\Cmf\Bundle\BlogBundle\Repository;
+
 use Doctrine\ODM\PHPCR\DocumentRepository;
-use Symfony\Cmf\Bundle\BlogBundle\Document\Post;
+use Doctrine\ODM\PHPCR\Mapping\ClassMetadata;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Cmf\Bundle\BlogBundle\Doctrine\Phpcr\Post;
 
 class PostRepository extends DocumentRepository
```
Comment (lsmith77) [inline Repository/PostRepository.php]: maybe lets do this in another PR after this one is merged?

File: Resources/config/cmf_routing_auto.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Symfony\Cmf\Bundle\BlogBundle\Doctrine\Phpcr\Blog:
+    uri_schema: /blogs/{blog}
```
Comment (lsmith77) [inline Resources/config/cmf_routing_auto.yml]: @briancappello please switch this to XML before we can merge this.

File: composer.json [hunk 3]
```diff
@@ -11,28 +11,38 @@
...
         "php": ">=5.3.3",
-        "symfony/framework-bundle": "~2.2",
-        "symfony-cmf/core-bundle": "1.0.*",
+        "symfony/framework-bundle": "~2.3",
+        "symfony-cmf/core-bundle": "1.2.*",
+        "symfony-cmf/content-bundle": "1.2.*",
         "symfony-cmf/routing-auto-bundle": "1.0.*",
-        "doctrine/phpcr-bundle": "1.0.*",
-        "doctrine/phpcr-odm": "1.0.*"
+        "sonata-project/doctrine-extensions": "~1.0.2",
```
Comment (lsmith77) [inline composer.json]: should this be ``~1.0.2|1.1`` ?

File: composer.json [hunk 4]
```diff
@@ -11,28 +11,38 @@
...
+        "doctrine/phpcr-bundle": "~1.2",
+        "doctrine/phpcr-odm": "~1.2",
+        "friendsofsymfony/rest-bundle": "~1.4.0"
```
Comment (lsmith77) [inline composer.json]: this should be ``~1.4``

File: composer.json [hunk 5]
```diff
@@ -11,28 +11,38 @@
...
     },
     "require-dev": {
-        "symfony-cmf/testing": "1.1.*",
-        "sonata-project/doctrine-phpcr-admin-bundle": "1.0.*",
-        "symfony-cmf/core-bundle": "1.0.*"
+        "phpunit/phpunit": "~3.7.28",
+        "symfony-cmf/testing": "1.2.*",
+        "sonata-project/doctrine-phpcr-admin-bundle": "1.2.*",
+        "doctrine/orm": "~2.4.7",
+        "nelmio/alice": "~1.7.2",
```
Comment (lsmith77) [inline composer.json]: should this be ``~1.7.2|1.8`` ?

File: composer.json (hunk 3 above)
Comment (dbu) [inline composer.json]: or `~1.0,>=1.0.2`?

File: composer.json (hunk 5 above)
Comment (WouterJ) [inline composer.json]: don't you mean: `~1.7,>=1.7.2`?

**Comment by lsmith77** [comment]
ping

**Comment by ConneXNL** [comment]
any update on this? im trying out symfony cmf but this is really discouriging

**Comment by WouterJ** [comment]
@connxnl keep in mind that this bundle is one of the showcase bundles of the CMF. It's goal is not so much of providing functionality, but more of providing examples of how to combine multiple CMF bundles.

Seems like this PR is ready to be merged, but we didn't get a notice of the update (maintainers only get a notice about comments). So thanks for your head up!

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,6 @@
 Tests/Resources/app/cache
 Tests/Resources/app/logs
+bin/
 composer.lock
-vendor
+vendor/
+.idea
```
Comment (dbu) [inline .gitignore]: this is the responsibility of the global git configuration of the developer, it is not specific to the project but to specific development environments. please remove this line.

**Comment by dbu** [comment]
@dantleech can you check if you see any blockers in here? i  can do some minimal cleanup, squash the commits and merge.

File: Controller/BaseController.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use FOS\RestBundle\View\View;
+
+/**
+ * Base Controller
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
```
Comment (dantleech) [inline Controller/BaseController.php]: I didn't write this :)

File: Controller/BaseController.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2014 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+
```
Comment (dantleech) [inline Controller/BaseController.php]: Extra line

File: DependencyInjection/CmfBlogExtension.php [hunk 9]
```diff
@@ -35,53 +35,57 @@ public function load(array $configs, ContainerBuilder $container)
...
+        $loader->load('admin.xml');
     }
 
-    private function loadMenuIntegration($config, XmlFileLoader $loader, ContainerBuilder $container)
+    protected function loadMenuIntegration(array $config, XmlFileLoader $loader, ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
-        if ('auto' === $config['integrate_menu']['enabled'] && !isset($bundles['CmfMenuBundle'])) {
+        if (!isset($bundles['CmfMenuBundle'])) {
             return;
```
Comment (dantleech) [inline DependencyInjection/CmfBlogExtension.php]: Menu integration is explicit .. so we should throw an Exception here if the Bundle is not registered

File: DependencyInjection/CmfBlogExtension.php [hunk 10]
```diff
@@ -35,53 +35,57 @@ public function load(array $configs, ContainerBuilder $container)
...
+            $this->loadPhpcrPersistence($config, $loader, $container);
+        }
+
+        if (isset($config['sonata_admin']) && $config['sonata_admin']['enabled']) {
             $this->loadSonataAdmin($config, $loader, $container);
         }
-        if ($config['integrate_menu']['enabled']) {
+
+        if (isset($config['integrate_menu']) && $config['integrate_menu']['enabled']) {
             $this->loadMenuIntegration($config, $loader, $container);
```
Comment (dantleech) [inline DependencyInjection/CmfBlogExtension.php]: Should probably pass `$config['integrate_menu']` here instead of the whole configuration array.

File: DependencyInjection/Configuration.php [hunk 11]
```diff
@@ -35,45 +35,63 @@ public function getConfigTreeBuilder()
...
-                    ->values(array(true, false, 'auto'))
-                    ->defaultValue('auto')
+
+                // admin
+                ->arrayNode('sonata_admin')
+                    ->canBeEnabled()
                 ->end()
 
+                // menu
                 ->arrayNode('integrate_menu')
```
Comment (dantleech) [inline DependencyInjection/Configuration.php]: I wonder if this should better be simply `menu`  or `cmf_menu`

File: Model/Post.php [hunk 12]
```diff
@@ -10,209 +10,252 @@
...
+     * Set publish end date
+     *
+     * @param \DateTime $publishEndDate
+     * @return Post
+     */
     public function setPublishEndDate(\DateTime $publishEndDate = null)
     {
         $this->publishEndDate = $publishEndDate;
+
+        return $this;
```
Comment (dantleech) [inline Model/Post.php]: We don't use fluid interfaces in other places in the CMF, not sure we want to start here.

Comment (WouterJ) [inline Model/Post.php]: IIRC, we do use it in a couple of places

File: Resources/config/services.xml [hunk 13]
```diff
@@ -0,0 +1,31 @@
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+<parameters>
+    <parameter key="cmf_blog.blog_controller.class">Symfony\Cmf\Bundle\BlogBundle\Controller\BlogController</parameter>
+    <parameter key="cmf_blog.post_controller.class">Symfony\Cmf\Bundle\BlogBundle\Controller\PostController</parameter>
+</parameters>
+
+<services>
```
Comment (dantleech) [inline Resources/config/services.xml]: Bad indentation

File: Tests/Functional/Admin/BlogAdminTest.php [hunk 14]
```diff
@@ -12,16 +12,68 @@
...
-        $client->request('GET', '/admin/dashboard');
-        $response = $client->getResponse();
-        $this->assertEquals(200, $response->getStatusCode());
+        $form = $crawler->selectButton('Create')->form();
+        $formPrefix = $this->getAdminFormNamePrefix($crawler);
+
+        $form->setValues(array(
+            $formPrefix.'[name]'           => 'Test Blog',
+            $formPrefix.'[description]'    => 'A blog lives here.',
+            $formPrefix.'[parentDocument]' => '/cms/blogs',
```
Comment (dantleech) [inline Tests/Functional/Admin/BlogAdminTest.php]: We don't do whitespace alignment generally.

File: composer.json [hunk 15]
```diff
@@ -11,28 +11,38 @@
...
     "require-dev": {
-        "symfony-cmf/testing": "1.1.*",
-        "sonata-project/doctrine-phpcr-admin-bundle": "1.0.*",
-        "symfony-cmf/core-bundle": "1.0.*"
+        "phpunit/phpunit": "~3.7.28",
+        "symfony-cmf/testing": "1.2.*",
+        "sonata-project/doctrine-phpcr-admin-bundle": "1.2.*",
+        "doctrine/orm": "~2.4.7",
+        "nelmio/alice": "~1.7,>=1.7.2",
+        "phpcr/phpcr-shell": "@alpha"
```
Comment (dantleech) [inline composer.json]: `@beta`

File: run_tests.sh [hunk 16]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+sh vendor/symfony-cmf/testing/bin/travis/doctrine_orm.sh
+sh vendor/symfony-cmf/testing/bin/travis/phpcr_odm_doctrine_dbal.sh
+bin/phpunit
```
Comment (dantleech) [inline run_tests.sh]: This can be moved to the `.travis` file

File: DependencyInjection/CmfBlogExtension.php (hunk 9 above)
Comment (dbu) [inline DependencyInjection/CmfBlogExtension.php]: this is removing the true|false|auto mechanism. i add it back in, we
want that for convenient developer experience.

File: composer.json (hunk 15 above)
Comment (dbu) [inline composer.json]: that should not be a dependency of the bundle, not even dev

**Comment by dbu** [comment]
cleaned up and created a new branch in this repo
